generator/internal/genclient/language/internal/rust/rust.go

```diff
@@ -662,20 +662,28 @@ func (c *Codec) PackageName(api *genclient.API) string {
 }
 
 func (c *Codec) validatePackageName(newPackage, elementName string) error {
-	if c.SourceSpecificationPackageName == "" {
-		c.SourceSpecificationPackageName = newPackage
+	if c.SourceSpecificationPackageName == newPackage {
 		return nil
 	}
-	if c.SourceSpecificationPackageName == newPackage {
+	// Special exceptions for mixin services
```

Comment:
This is not the first one, but we have some duplicate code rust.go and golang.go.  We should consider some common codec things.

Reply:
ack, agreed: https://github.com/googleapis/google-cloud-rust/issues/294